server.js

```diff
@@ -44,18 +72,8 @@ function handleSocketConnection(socket) {
     io.to(currentRoom).emit('lobbyUpdate', users.list())
 
     if (users.count() >= MIN_PLAYER_COUNT) {
-      console.log(`${currentRoom} will start a game in ${START_GAME_DELAY} sec`)
-      clearTimeout(countDownTimer)
-      countDownTimer = setTimeout(function startAGame() {
-        if (users.count() >= MIN_PLAYER_COUNT) { // if nobody has left lobby
-          io.to(currentRoom).emit('stateChange', 'game')
-          console.log(`${currentRoom} started a game!`)
-          currentRoom = generateRoomName()
-          users.clear()
-          io.emit('lobbyUpdate', users.list())
-        }
-      }, START_GAME_DELAY * ONE_SECOND + DELAY_BUFFER_IN_MS)
-      io.to(currentRoom).emit('countdown', START_GAME_DELAY)
+      startGame()
+      assignRoles(currentRoom)
```

Comment:
Put it within `startGame()`, but only call it after the game has actually started, not when we've started the countdown to a game. You're just calling it in the wrong place.